File: chapter-5_arrays/python/one-dimensional-array.py
import math

numList = [0, 1, 1, 2, 3, 5, 8, 13, 21]

print(numList)

min = 0
max = 0

for value in numList:
    if value < min:
        min = value
    if value > max:
        max = value

print('min: {}, max: {}'.format(min, max))

total = 0

for value in numList:
    total += value;

print('total: {}'.format(total))

average = total / len(numList)

print('average: {}'.format(average))

variance = 0

for value in numList:
    variance += math.pow(value - average, 2)

variance /= len(numList)

print('variance: {}'.format(variance))

stanDev = math.sqrt(variance)

print('standard deviation: {}'.format(stanDev))

File: chapter-5_arrays/javascript/one-dimensional-array.js
'use strict';

var array = [0, 1, 1, 2, 3, 5, 8, 13, 21];

console.log(array);

//Min and Max:
var min = array[0];
var max = array[0];
for(var i = 1; i < array.length; i++) {
  if(array[i] < min) {
    min = array[i];
  }
  if(array[i] > min) {
    max = array[i];
  }
}

console.log('min: ' + min + ', max: ' + max);

//Total:
var totalSum = 0;
for (var i = 0; i < array.length; i++) {
  totalSum += array[i];
}

console.log('total: ' + totalSum);

//Mean (Average):
var average = totalSum / array.length;

console.log('average: ' + average);

//Variance: the avergae of the sums of the squares of the values - the mean:
var variance = 0;
for (var i = 0; i < array.length; i++) {
  variance += Math.pow(array[i] - average, 2);
}
variance /= array.length;

console.log('variance: ' + variance);

//standard deviation: square root of the variance:
var stanDev = Math.sqrt(variance);

console.log('standard deviation: ' + stanDev);
